lib/stat_tracker.rb

```diff
@@ -87,22 +87,36 @@ def count_of_teams
        @teams_stats_data.size
     end
    
-    def best_offense
-
-        teams_goals = Hash.new(0)
-        games_played = Hash.new(0)
+    def team_seasons_goals(location)
+        
+        teams_data = Hash.new { |hash, key| hash[key] = { goals: 0, games_played: 0 } }
 
         @seasons_stats_data.each do |game_id, season|
-            teams_goals[season.team_id] += season.goals
-            games_played[season.team_id] += 1
+            if location.include?(season.hoa)
```

Comment:
I like the use of include?, my first instinct would be to have a complicated if statement using ==, but I see the logic and efficiency of your approach.